src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/OAuthCredmon.py

```diff
@@ -151,43 +134,26 @@ def refresh_access_token(self, username, token_name):
         else:
             return True
 
-    def delete_tokens(self, username, token_name):
-        exts = ['.top', '.use', '.meta', '.mark']
-        base_path = os.path.join(self.cred_dir, username, token_name)
-
-        success = True
-        for ext in exts:
-            if os.path.exists(base_path + ext):
-                try:
-                    os.unlink(base_path + ext)
-                except OSError as e:
-                    self.log.debug('Could not remove %s: %s', base_path + ext, e.strerror)
-                    success = False
-            else:
-                self.log.debug('Could not find %s', base_path + ext)
-        return success
-
 
     def check_access_token(self, access_token_path):
 
         (basename, token_filename) = os.path.split(access_token_path)
         (cred_dir, username) = os.path.split(basename)
         token_name = os.path.splitext(token_filename)[0] # strip .use
 
+        if self.providers and (token_name not in self.providers) and ('*' not in self.providers):
+            if htcondor and (token_name + "CLIENT_ID") not in htcondor.param:
+                self.log.warning(f"User requested token provider {token_name} but parameter {token_name}_CLIENT_ID not set in config; ignoring request")
+            return
+        elif htcondor and (token_name + "CLIENT_ID") not in htcondor.param:
```

Comment:
```suggestion
            if htcondor and (token_name + "_CLIENT_ID") not in htcondor.param:
                self.log.warning(f"User requested token provider {token_name} but parameter {token_name}_CLIENT_ID not set in config; ignoring request")
            return
        elif htcondor and (token_name + "_CLIENT_ID") not in htcondor.param:
```

Reply:
If you're going to have to do this, maybe use `get_credmon_config()`?

Reply:
`get_credmon_config()` is not defined for the `OAuthCredmon`, but looking over this block again, it definitely needed some cleanup/explanation, done in daa4a4706b

Reply:
Looks good to me.